mk/PcapPlusPlus.mk.linux

```diff
@@ -3,8 +3,18 @@
 # includes
 PCAPPP_INCLUDES += -I/usr/include/netinet
 
+ifdef BUILD_WITH_STATIC_LIBPCAP
+PCAPPP_INCLUDES += -I$(STATIC_LIBPCAP_INCLUDE)
+endif
+
 # libs
-PCAPPP_LIBS += -lpcap -lpthread
+PCAPPP_LIBS += -lpthread
+
+ifndef BUILD_WITH_STATIC_LIBPCAP
+PCAPPP_LIBS += -lpcap
+else
+PCAPPP_LIBS += $(STATIC_LIBPCAP_LIB)
+endif
```

Comment:
generally speaking, I try to avoid `ifdef` statements in `PcapPlusPlus.mk.*` files because they make the build system even more complicated. Instead those `ifdef` statements usually reside in the `configure-*` files. Since I'm not sure there is an easy way to replace `-lpcap` with the static libpcap lib without using these `ifdefs` my suggestion would be to remove the feature of building statically with libpcap.

Is there a way to avoid static linking in OSSFuzz?

Otherwise we can think of a way to create a separate `PcapPlusPlus.mk.linux.fuzzing` and maybe create a command line flag in `configure-linux.sh` that will require the user to provide static libpcap location only if he uses fuzzing

Reply:
Sadly, there's no way to avoid static linking in OSSFuzz, as explained [here](https://github.com/google/oss-fuzz/issues/25#issuecomment-254320632).

I thought it would be nice to have it across all the library, tests, and examples, but if you prefer to keep it only for the fuzzers I'm fine with it too. I will try to do this today.

Reply:
I pushed a different version that removes the `ifdef` statements from `PcapPlusPlus.mk.*` platform specific files. I created alternate versions for each platform file such as `PcapPlusPlus.mk.linux.static_libpcap` with a check on the `configure-*.sh` files to write one or another to the final `PcapPlusPlus.mk` file.

With this I'm preserving the functionality of allowing to build everything with static libpcap while keeping the `PcapPlusPlus.mk.{linux,macosx,freebsd}` files clean of `ifdef` statements.

Reply:
while that's a valid solution, it add some complexity to the build system because from now on each parameter/flag we add to `PcapPlusPlus.mk.*` we'll need to add it also in the corresponding `PcapPlusPlus.mk.*.static_libpcap`. I'm afraid this will be hard to maintain.

My suggestion would be to create a new `configure-fuzzing.sh` script which will incorporate everything that is needed to compile PcapPlusPlus with fuzzing on Linux and create separate `PcapPlusPlus.mk.fuzzing` that will contain the build flags etc.

This will decouple fuzzing from everything else in PcapPlusPlus and will be easier to maintain. Please let me know what you think

Reply:
I'm good with that too if you don't want to keep the ability of linking PcapPlusPlus against static libpcap for all other builds.

Will do the decoupling today.